servo/connectors/wavefront.py

```diff
@@ -0,0 +1,353 @@
+import asyncio
+import datetime
+import functools
+import re
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import httpcore._exceptions
+import httpx
+import pydantic
+
+import servo
+
+DEFAULT_BASE_URL = "http://wavefront.com:2878"
+API_PATH = "/api/v2/"
+
+
+class WavefrontMetric(servo.Metric):
+    """WavefrontMetric objects describe metrics that can be measured by querying
+    Wavefront.
+    """
+
+    query: str
+    """A WQL query that returns the value of the target metric.
+
+    For details on Wavefront, see the [Wavefront
+    Querying](https://docs.Wavefront.com/query_language_getting_started.html)
+    documentation.
+    """
+
+    granularity: pydantic.constr(
+        regex=r"^[smhd]"
+    ) = "m"
+
+    """The resolution of the query. The granularity resolution determines the number of data points captured across a
+    query range. A query's granularity is completely independent from any range durations specified in the WQL expression it evaluates.
+    
+    Available values: s, m, h, d.
+    """
```

Comment:
I'd probably argue for turning this into an enum as discussed below that Pydantic will constrain it for us and its self-documenting what the available values are.